iceoryx_utils/include/iceoryx_utils/design_pattern/creation.hpp

```diff
@@ -23,6 +23,68 @@
 
 namespace DesignPattern
 {
+/// @brief This pattern can be used if you write an abstraction where you have
+///        to throw an exception in the constructor when you for instance would
+///        like to manage a resource and the constructor was unable to acquire
+///        that resource.
+///        In this case you inherit from Creation and your class has two more
+///        static factory methods - create and verify.
+///        `create` forwards all arguments to the underlying class constructor and
+///        if the construction was successful an expected containing the type is
+///        returned, otherwise an error value which describes the error.
+/// @code
+///   enum class MyResourceAbstractionError {
+///     ResourceNotAvailable,
+///     BlaBlubError
+///   };
+///   class MyResourceAbstraction : public Creation<MyResourceAbstraction, MyResourceAbstractionError> {
+///     public:
+///       // some public methods
+///       MyResourceAbstraction & operator=(MyResourceAbstraction && rhs) noexcept {
+///         if ( this != &rhs ) {
+///           // always call the creation move assignment operator when you have a user defined
+///           // move operation
+///           CreationPattern_t::operator=(std::move(rhs));
+///
+///           // user move code
+///         }
+///         return *this;
+///       }
+///
+///       // the creation pattern is the only one which should be allowed to construct
+///       // the class, therefore it has to be friend of that class
+///       friend class Creation<MyResourceAbstraction, MyResourceAbstractionError>;
+///     private:
+///       MyResourceAbstraction(int a) {
+///         if ( a > 0) {
+///           // we are able to initialize the class an set m_isInitialized to true
+///           m_isInitialized = true;
```

Comment:
`m_isInitialized` is not declared because it is provided by the `Creation` class. We should maybe explicitly state that the user need to set this bool when the constructor completes successfully.